pkg/syncer/specsyncer.go

```diff
@@ -43,12 +45,38 @@ func deepEqualApartFromStatus(oldObj, newObj interface{}) bool {
 	if !isOldObjUnstructured || !isNewObjUnstructured {
 		return false
 	}
-	if !equality.Semantic.DeepEqual(oldUnstrob.GetAnnotations(), newUnstrob.GetAnnotations()) {
+
+	// TODO(jmprusi): Remove this after switching to virtual workspaces.
+	// remove status annotation from oldObj and newObj before comparing
+	oldAnnotations := oldUnstrob.GetAnnotations()
+	for k := range oldAnnotations {
+		if !strings.HasPrefix(k, LocationStatusAnnotationName("")) {
+			delete(oldAnnotations, k)
```

Comment:
In this case, we need to remove from the comparison the status annotations to avoid reconcile loops, I'm not "setting" those annotations back into the Object (SetAnnotations()) and NestedStringMap() (used by GetAnnotations() seems to return a copy the annotations)

Reply:
`GetAnnotations` does not guarantee that. Look at `metav1.Object`:
```
func (meta *ObjectMeta) GetAnnotations() map[string]string            { return meta.Annotations }
```
We cannot depend on that behaviour of unstructured. Just call `NestedStringMap` manually. Then it doesn't work here by coincidence.

Reply:
Thanks for this explanation @sttts . Useful.